v2/yarpcrouter/router.go

```diff
@@ -58,19 +59,78 @@ type MapRouter struct {
 func NewMapRouter(defaultService string) MapRouter {
 	return MapRouter{
 		defaultService:            defaultService,
-		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.Procedure),
+		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.TransportProcedure),
 		serviceNames:              map[string]struct{}{defaultService: {}},
 	}
 }
 
+// MapToTransportProcedures converts encoding-level procedures to transport-level procedures
+func MapToTransportProcedures(encodingProcedures []yarpc.EncodingProcedure) []yarpc.TransportProcedure {
+	transportProcedures := []yarpc.TransportProcedure{}
+	for _, p := range encodingProcedures {
+		var transportHandlerSpec yarpc.TransportHandlerSpec
+		switch p.HandlerSpec.Type() {
+		case yarpc.Unary:
+			transportHandlerSpec = yarpc.NewUnaryTransportHandlerSpec(unaryTransportHandler{p})
+		default:
+			panic(p.HandlerSpec.Type())
+		}
+
+		transportProcedures = append(
+			transportProcedures,
+			yarpc.TransportProcedure{
+				Name:        p.Name,
+				Service:     p.Service,
+				HandlerSpec: transportHandlerSpec,
+				Encoding:    p.Encoding,
+				Signature:   p.Signature,
+			},
+		)
+	}
+
+	return transportProcedures
+}
+
+// NewMapRouterWithProcedures constructs a new MapRouter with the given default service name and registers
+// the given transport-level procedures
+func NewMapRouterWithProcedures(defaultService string, transportProcedures []yarpc.TransportProcedure) MapRouter {
+	router := NewMapRouter(defaultService)
+	router.Register(transportProcedures)
+	return router
+}
+
+// Allows encoding-level procedures to act as transport-level procedures
+type unaryTransportHandler struct {
+	h yarpc.EncodingProcedure
+}
```

Comment:
This is a little confusing - the transport handler has an `EncodingProcedure`. I would expect a `TransportProcedure` here. If this is intentional, could you clarify this with a comment somewhere?